Scripts/Runtime/Entities/TaskDriver/AbstractTaskDriver.cs

```diff
@@ -91,30 +94,30 @@ bool ITaskSetOwner.HasCancellableData
             }
         }
 
-        protected ITaskDriverSystem System
-        {
-            get => new ContextTaskDriverSystemWrapper(TaskDriverSystem, this);
-        }
-
         /// <summary>
         /// Creates a new instance of a <see cref="AbstractTaskDriver"/>
         /// </summary>
         /// <param name="world">The <see cref="World"/> this Task Driver is a part of.</param>
-        /// <param name="uniqueMigrationSuffix">
-        /// An optional unique suffix to identify this TaskDriver by. This is necessary when there are two or more of the
+        /// <param name="parent">The parent <see cref="AbstractTaskDriver"/> if it exists</param>
+        /// <param name="uniqueContextIdentifier">
+        /// An optional unique identifier to identify this TaskDriver by. This is necessary when there are two or more of the
         /// same type of TaskDrivers at the same level in the hierarchy.
         /// Ex.
         /// ShootTaskDriver
         ///  - TimerTaskDriver (for time between shots)
         ///  - TimerTaskDriver (for reloading)
         /// 
         /// Both TimerTaskDriver's would conflict as being siblings of the ShootTaskDriver so they would need a unique
-        /// migration suffix to distinguish them for ensuring migration happens properly between worlds.
+        /// context identifier to distinguish them for ensuring migration happens properly between worlds and data
+        /// goes to the correct location.
         /// </param>
-        protected AbstractTaskDriver(World world, string uniqueMigrationSuffix = null)
+        protected AbstractTaskDriver(World world, AbstractTaskDriver parent = null, string uniqueContextIdentifier = null)
         {
-            m_UniqueMigrationSuffix = uniqueMigrationSuffix ?? string.Empty;
             World = world;
+            Parent = parent;
+            Parent?.m_SubTaskDrivers.Add(this);
```

Comment:
Fair, I've added it as a TODO in #241 

I can't just lazy create because it will get called right away in the constructor when we register with the TaskDriverManagementSystem.

Re-generating the ID is an option that I just tried  but it would require cascading down through all the data as well and regenerating their ID's and then removing all the old IDs from the lookups and replacing with the new ones. 

I thin kas part of #241 I'll need to either have an efficient two pass system or more likely, have the concept of local ids vs the world ids. 

The data will have an id that is relative to their data owner (TaskDriver etc).
Then during hardening, the world lookups will get built which will concat to generate the world unique id based on the full path.

Reply:
It's specifically that add back to the parent private list that gets me. With a bit of brainstorming I think we can fix that stinky bit with a smart pattern.

Reply:
Yep that's fair. #241 will likely make it irrelevant and a nicer pattern too.